Freshen string comparison code (#44736)
dotnet/docs: docs/csharp/how-to/search-strings.md

The search pattern describes the text you search for. The following table describes each element of the search pattern. (The following table uses the single `\`, which must be escaped as `\\` in a C# string).

| Pattern  | Meaning                          |
|----------|----------------------------------|
| `the`    | match the text "the"             |
| `(eir)?` | match 0 or 1 occurrence of "eir" |
| `\s`     | match a white-space character    |

:::code language="csharp" interactive="try-dotnet-method" source="./snippets/strings/SearchStrings.cs" id="Snippet3":::

> [!TIP]
> The `string` methods are generally better choices when you're searching for an exact string. Regular expressions are better when you're searching for some pattern in a source string.

## Does a string follow a pattern?

> The following code uses regular expressions to validate the format of each string in an array. The validation requires that each string has the form of a telephone number: three groups of digits separated by dashes where the first two groups contain three digits and the third group contains four digits. The search pattern uses the regular expression `^\\d{3}-\\d{3}-\\d{4}$`. For more information, see [Regular Expression Language - Quick Reference](../../standard/base-types/regular-expression-language-quick-reference.md).

Reviewer: ```suggestion
The following code uses regular expressions to validate the format of each string in an array. The validation requires that each string is formatted as a telephone number: three groups of digits separated by dashes where the first two groups contain three digits and the third group contains four digits. The search pattern uses the regular expression `^\\d{3}-\\d{3}-\\d{4}$`. For more information, see [Regular Expression Language - Quick Reference](../../standard/base-types/regular-expression-language-quick-reference.md).
```
